tests/de.rs

```diff
@@ -355,3 +355,13 @@ fn deserializes_ignored() {
 
     deserializes_to("{ a: 1, ignored: 42, b: 2 }", S { a: 1, b: 2 });
 }
+
+#[test]
+fn deserializes_json_values() {
+    // As int if json uses int type
+    deserializes_to("0x2A", serde_json::json!(42));
+    deserializes_to("42", serde_json::json!(42));
+
+    // As float if json calls for explicit float type
+    deserializes_to("42.", serde_json::json!(42.));
```

Comment:
I can see one more case that needs to be handled:

```rust
deserializes_to("4e2", serde_json::json!(4e2));
```

Checking for `e` and `E` in `is_int` (but making sure it isn't a hex literal) should do the trick.

Reply:
✅
Good point, done!